File: app/src/main/java/com/example/win/easy/repository/web/request/BackendRequestService.java
package com.example.win.easy.repository.web.request;

import com.example.win.easy.repository.web.dto.SongListDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface BackendRequestService {

    @POST("resource/uid/phone")
    Call<String> getUidByPhone(@Query("account") String phone, @Query("password") String password);

    @POST("resource/uid/email")
    Call<String> getUidByEmail(@Query("account")String email,@Query("password")String password);

    @GET("resource/songlist/uid")
    Call<List<SongListDTO>> getAllSongListsByUid(@Query("uid") String uid);

    @POST("resource/songlist/phone")
    Call<List<SongListDTO>> getAllSongListsByPhone(@Query("account")String phone,@Query("password")String password);

    @POST("resource/songlist/email")
    Call<List<SongListDTO>> getAllSongListsByEmail(@Query("account")String email,@Query("password")String password);

}

File: app/src/main/java/com/example/win/easy/web/request/BackendRequestService.kt
package com.example.win.easy.web.request

import com.example.win.easy.web.dto.SongDTO
import com.example.win.easy.web.dto.SongListDTO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface BackendRequestService {
    @POST("uid/phone")
    suspend fun getUidByPhone(@Query("account") phone: String, @Query("password") password: String): String

    @POST("uid/email")
    suspend fun getUidByEmail(@Query("account") email: String, @Query("password") password: String): String

    @GET("songlist/uid")
    suspend fun getAllSongListsByUid(@Query("uid") uid: String): List<SongListDTO>

    @GET("song/list")
    suspend fun getSongsBySongListId(@Query("songListId") songListId: String): List<SongDTO>
}